### PR TITLE
Adding default for read timeout in the event collection

### PR DESCRIPTION
**Pull request by CharlyF**
### What does this PR do?

Event collection read timeout was parsed from the env/the datadog general config.
It should be in the yaml of the check.

This also defaults it at 1 second.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: pkg/util/kubernetes/apiserver/events.go [hunk 1]
```diff
@@ -27,9 +27,15 @@ func (c *APIClient) RunEventCollection(resVer string, lastListTime time.Time, ev
 	syncTimeout := time.Duration(resync) * time.Second
 	// list if latestResVer is "" or if lastListTS is > syncTimeout
 	diffTime := time.Now().Sub(lastListTime)
-	if resVer == "" || diffTime > syncTimeout {
-		log.Debugf("return listForEventResync diffTime: %d > %d", diffTime, syncTimeout)
-		return c.listForEventResync(eventReadTimeout, eventCardinalityLimit, filter)
+	if resVer == "" || time.Now().Sub(lastListTime) > syncTimeout {
```
Comment (clamoriniere) [inline pkg/util/kubernetes/apiserver/events.go]:
```suggestion
	if resVer == "" || diffTime > syncTimeout {
```

File: pkg/collector/corechecks/cluster/kubernetes_apiserver.go [hunk 2]
```diff
@@ -31,11 +31,12 @@ import (
...
-	kubernetesAPIServerCheckName = "kubernetes_apiserver"
-	eventTokenKey                = "event"
-	maxEventCardinality          = 300
-	defaultResyncPeriodInSecond  = 300
+	KubeControlPaneCheck          = "kube_apiserver_controlplane.up"
+	kubernetesAPIServerCheckName  = "kubernetes_apiserver"
+	eventTokenKey                 = "event"
+	maxEventCardinality           = 300
+	defaultResyncPeriodInSecond   = 300
+	defaultTimeoutEventCollection = 1000
```
Comment (clamoriniere) [inline pkg/collector/corechecks/cluster/kubernetes_apiserver.go]: could we set it a 2000ms here and remote the `*2` in the ticker creation

```suggestion
	defaultTimeoutEventCollection = 2000
```

Comment (CharlyF) [reply pkg/collector/corechecks/cluster/kubernetes_apiserver.go]: ok

**Review by clamoriniere — APPROVED** [review]

**Review by clamoriniere — DISMISSED** [review]

**Review by clamoriniere — APPROVED** [review]

**Pull request merged**